### ISSUE TITLE
Binding a NaN value to the v-model of a FormKit component causes a "maximum recursive updates exceeded" error.

### ISSUE THREAD

**Issue by wangxdmm** [formkit/formkit#1555]
### Reproduction

https://formkit.link/1e3a5f0e3c1ac306be4eaa1e2c344c01

### Describe the bug

I think it maybe `eq` function at `packages/utils/src/index.ts` that fails to handle the equality of `NaN` correctly.

```ts
expect(eq(NaN, NaN)).toBe(true) // error
```

### Environment

• OS: WSL: Ubuntu-22.04.3
• Browser chrome

formkit
```
"@formkit/core": 1.6.7
"@formkit/vue": 1.6.7
"@formkit/inputs": 1.6.7
"@formkit/rules": 1.6.7
"@formkit/validation": 1.6.7
```

**Comment by justin-schroeder** [comment]
Well, to be clear, thats just javascript:

```
Number.NaN === Number.NaN
// false
```

**Comment by wangxdmm** [comment]
> Well, to be clear, thats just javascript:
> 
> ```
> Number.NaN === Number.NaN
> // false
> ```

Yes, I know that 's  just JavaScript, but as a form value, I think `NaN` can actually be considered the same value.

In fact, the `eq` function has no issues, but when handling Vue's model, Vue's `computed` use `Object.is` to compare, treating NaN as the same value. So, could you consider handling this NaN case?

**Comment by justin-schroeder** [comment]
Fair enough, we can do that.
